### ISSUE TITLE
Feature request: Read and display the computer description from Windows machines

### ISSUE THREAD

**Issue by tylla** [Ylianst/MeshCentral#6556]
HI!
All windows computers have a "Computer description" field that can be set when one renames the computer. Actually Linux Samba has a similar "server string" entry. This field allows to enter extra information about the machine, and is some times unpopulated or contains some non-sense like "Kitchen computer", but it can be very well used for some really useful things like for example information about the location and/or user of the computer.

Currently MeshCentral doesn't use this value but it would be really nice to extract it and show it in the "My Devices" list and on the General tab of the machine overview.
Currently there is an "OS Name" and there could be an "OS Description" as well which would show this value.

If one would like to check the current value:
From the Windows GUI:
1. Open File Explorer,
2. Right-click This PC and choose Properties, and
3. Click Change settings on the right 

From the command line (cmd.exe):
- `wmic os get description`
or
- `reg query "HKLM\SYSTEM\CurrentControlSet\services\LanmanServer\Parameters" /v srvcomment`

I tried to search for this feature in the open/closed issues, but couldn't find anything similar, I hope I didn't missed it. :)

Please indicate if I should provide any additional information, which I would happily provide.

**Comment by si458** [comment]
Ah right, u are talking about the old style `Computer Description` inside the system properties panel!
![Screenshot_20241125_165207_Firefox](https://github.com/user-attachments/assets/6b4510aa-6dc1-404d-b74f-cd1738125b9b)

**Comment by si458** [comment]
if you want me to add it into the `Details` tab,
i can add an entry under `Operating System` below `Name` saying `Description`
this way we can expand on it in future for other OS like linux

adding it into the General tab and then the list view is more fiddly because we already have `Description`
so it would cross the 2 over, ill have to think on it

**Comment by tylla** [comment]
Hmm, the `Details `tab is a fit place right under the `Name` entry.
OTOH on the `General` tab, an `OS Description` would be natural as well.
Actually if it's not a too big hurdle, I would be really happy to see it in both places as well. As the `Name` in the `Details` tab is listed in the `General` tab as `OS Name`, the same logic could be applied: `Description` - `OS Description`.

**Comment by si458** [comment]
OK leave it with me to have a play with!
My thinking is we display OS description ONLY if u actually have one, if its blank then don't display it :)

**Comment by si458** [comment]
so this is already included and u can access it using the websocket api (node.hardware.windows.osinfo.Descrption)
we just dont display it on the web ui, who knew haha!

i have included it in the Details tab only because the General tab uses a different set of results
and this would mean pushing i would have to query the info AGAIN just to display it in the General Tab which isnt ideal!

**Comment by tylla** [comment]
Wow, thank you very much. That was some real quick thing. I can't wait to test it.

I don't want to sound too demanding :), but is there a chance that at some point in the future the General tab will get this as well?

**Comment by si458** [comment]
yes i will look at it in future but at the minute the way the code is setup, is basically
all the details tab is done when the meshagent starts up and then every 15mins
where as the general tab can change on the fly
so in a sense it would querying the twice which isnt idea!

IF you brave enough u can test it now using the development build (no translations tho so only english!)
1. stop meshcentral,
2. BACKUP ALL FOLDERS!!!
3. `npm install Ylianst/MeshCentral`
4. start meshcentral

or docker master images (translations included)
`ghcr.io/ylianst/meshcentral:master`

**Comment by tylla** [comment]
Ohh, that's perfect, I will wait with patience.

Well, it's a production environment, so I'm not that brave. But we will see it surely soon. :)

Thanks again!

**Comment by si458** [comment]
this was added in 1.1.37!
well it was added into the details tab!
still working on other places for you!

**Comment by tylla** [comment]
Sorry for the late comeback, we finally updated our system and tested the `Details` tab. It works as it should, thank you again.
Please notify if you found some solution for the `General` tab.

**Comment by tylla** [comment]
It would be really nice to have in the General tab and between the columns of the list view, but I can live with that if there is no time for this.
I'm already grateful for the Details tab... :)

**Comment by tylla** [comment]
I guess we should still keep this open for the other parts?!

**Comment by tylla** [comment]
No, not stale, only sleeping! :)

**Comment by si458** [comment]
<img width="576" height="163" alt="Image" src="https://github.com/user-attachments/assets/3da170d8-6e63-40a6-8198-79c1ea23fb1b" />

this already exists in the Details tab? was i missing it elsewhere?

**Comment by tylla** [comment]
Yepp, it is, and it is working flawlessly.
As a bonus, we talked about a possible addition to the `General` tab and as an extra column to the `Devices` list, but the different life cycle of those pages/fields made it harder to implement.

**Comment by si458** [comment]
#SIMONSTODOLIST
